screenapiexample/src/androidTest/java/net/gini/android/vision/screen/testhelper/PermissionsHelper.java

```diff
@@ -0,0 +1,85 @@
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.InstrumentationRegistry.getTargetContext;
+
+import android.Manifest;
+import android.annotation.TargetApi;
+import android.app.UiAutomation;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.os.ParcelFileDescriptor;
+import android.support.annotation.NonNull;
+import android.util.Log;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class PermissionsHelper {
+
+    private static final String LOG_TAG = "GrantPermission";
+
+    public static void grantExternalStoragePermission() throws InterruptedException {
+        grantExternalStoragePermission(getTargetContext().getPackageName());
+    }
+
+    public static void grantExternalStoragePermission(final String packageName)
+            throws InterruptedException {
+        grantPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE, packageName);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+            grantPermission(Manifest.permission.READ_EXTERNAL_STORAGE, packageName);
+        }
+    }
+
+    public static void grantCameraPermission() throws InterruptedException {
+        grantCameraPermission(getTargetContext().getPackageName());
+    }
+
+    public static void grantCameraPermission(final String packageName) throws InterruptedException {
+        grantPermission(Manifest.permission.CAMERA, packageName);
+    }
+
+    private static void grantPermission(final String permission, final String packageName) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if (hasPermission(permission, packageName)) {
+                return;
+            }
+            String command = "pm grant " + packageName + " " + permission;
+            executeShellCommand(command);
+        }
+    }
+
+    private static boolean hasPermission(String permission, String packageName) {
+        int checkResult = getTargetContext().getPackageManager().checkPermission(permission,
+                packageName);
+        return checkResult == PackageManager.PERMISSION_GRANTED;
+    }
+
+    /**
+     * Executes a command on the shell using {@link UiAutomation#executeShellCommand(String)}. It
+     * blocks until the standard out was read.
+     *
+     * @param command command to execute on the shell
+     * @return command's standard out
+     */
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    @NonNull
+    private static String executeShellCommand(final String command) {
+        UiAutomation uiAutomation = getInstrumentation().getUiAutomation();
+        try (ParcelFileDescriptor parcelFd = uiAutomation.executeShellCommand(command);
+             FileReader fileReader = new FileReader(parcelFd.getFileDescriptor());
+             BufferedReader reader = new BufferedReader(fileReader)) {
+            StringBuilder stringBuilder = new StringBuilder();
```

Comment:
Has this the correct indentation? It is not easy to read.

Reply:
Strangely it has correct indentation... due to the [try with resources statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) needing multiple lines the indentation looks off. Actually this is a case where curly braces on a new line would make it clearer 🤔...